### PR TITLE
provide the option to use 'acl bucket owner full control' option with the S3Service

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

provide the option to use 'acl bucket owner full control' option with the S3Service

## What is the intent behind these changes?

allows us to write to the ndmis reporting bucket

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt [hunk 1]
```diff
@@ -24,8 +25,14 @@ class S3Service {
     val objectRequest = PutObjectRequest.builder()
       .bucket(bucket.bucketName)
       .key(key)
+      .apply {
+        if (bucketOwnerFullControl) {
+          acl(ObjectCannedACL.BUCKET_OWNER_FULL_CONTROL)
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt:30]: Should this get a unit test?

**Review by sldblog — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt (hunk 1 above)
Comment (tomsmyers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt]: i actually changed the signature slightly - so you just pass the ACL directly. i can add a test to ensure it's appliied properly.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt [hunk 2]
```diff
@@ -24,8 +25,10 @@ class S3Service {
     val objectRequest = PutObjectRequest.builder()
       .bucket(bucket.bucketName)
       .key(key)
+      .apply { acl?.let { acl(it) } }
       .build()
 
+
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt:31]: This seems to displease `:ktlintMainSourceSetCheck`

**Review by sldblog — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/S3Service.kt]: Pushed e6d7113b3d0dfe021865d69bc6fe415675e66d4c to fix

**Pull request merged**